File: UserDetailsApi/Models/TaskModel.cs
using System.ComponentModel.DataAnnotations;
using UserDetailsApi.Enums;

namespace UserDetailsApi.Models
{
    public class TaskModel
    {
        public int Id { get; set; }

        [Required]
        public string Title { get; set; }

        public string Description { get; set; } = string.Empty;

        [Required]
        public Status Status { get; set; }

        [Required]
        public Priority Priority { get; set; }

        public DateTime? DueDate { get; set; }

        public DateTime CreatedAt { get; set; } = new DateTime();

    }
}

File: UserDetailsFrontend/src/app/core/models/task.model.ts
import { Priority } from "../enums/priority.enum";
import { Status } from "../enums/status.enum";

export class Task {
    id: Number = 0;
    title: string = ''
    description: string = '';
    status: Status = Status.ToDo
    priority: Priority = Priority.Low;
    dueDate?: Date = undefined;
    createdAt: Date = new Date();
}